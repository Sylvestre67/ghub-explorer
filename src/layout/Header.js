import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core';

import Grid from '@material-ui/core/Grid';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const isEven = process.env.REACT_APP_IS_EVEN;

const styles = theme => ({
	grow: {
		flexGrow: 1,
	},
	img: {
		width: '50px',
		height: 'auto',
	},
	bar: {
		display: 'flex',
		justifyContent: 'center',
	},
});

export function Header(props) {
	const { classes } = props;

	return (
		<AppBar position="relative" elevation={0}>
			<Toolbar className={classes.bar}>
				<Grid container>
					<Grid item xs={12}>
						{isEven ? (
							<div>
								<img
									className={classes.img}
									src={`${
										process.env.PUBLIC_URL
									}/img/even.png`}
									alt="even"
								/>
							</div>
						) : (
							<Typography
								variant="h6"
								color="inherit"
								className={classes.grow}>
								GHub Explorer
							</Typography>
						)}
					</Grid>
				</Grid>
			</Toolbar>
		</AppBar>
	);
}

Header.propTypes = {
	classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Header);
