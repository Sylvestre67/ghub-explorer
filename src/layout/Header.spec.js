import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';
let component;

const props = {
	classes: {
		grow: '',
		img: '',
	},
};

describe('Header', () => {
	beforeAll(() => {
		component = shallow(<Header {...props} />);
	});

	it('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});
});
