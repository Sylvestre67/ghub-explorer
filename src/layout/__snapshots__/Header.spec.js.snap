// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<WithStyles(AppBar)
  elevation={0}
  position="relative"
>
  <WithStyles(Toolbar)>
    <WithStyles(Grid)
      container={true}
    >
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <WithStyles(Typography)
          className=""
          color="inherit"
          variant="h6"
        >
          GHub Explorer
        </WithStyles(Typography)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
